addons/calendar/static/src/js/configure_external_sync_wizard_form_renderer.js

```diff
@@ -0,0 +1,87 @@
+/** @odoo-module **/
+
+import { FormRenderer } from "@web/views/form/form_renderer";
+import { useService } from "@web/core/utils/hooks";
+import { session } from "@web/session";
+
+const CALENDAR_CONFIGS = {
+    google: {
+        caption: 'Google',
+        restart_sync_method: 'restart_google_synchronization',
+        sync_route: '/google_calendar/sync_data',
+    },
+    microsoft: {
+        caption: 'Outlook',
+        restart_sync_method: 'restart_microsoft_synchronization',
+        sync_route: '/microsoft_calendar/sync_data',
+    }
+};
+
+
+export class ConfigureExternalSyncWizardFormRenderer extends FormRenderer {
+    setup() {
+        super.setup();
+        this.orm = useService('orm');
+        this.rpc = useService('rpc');
+    }
+
+    onConnect(ev) {
+        ev.preventDefault();
+        ev.stopImmediatePropagation();
+        const self = this;
+        const data = this.props.record.data;
+        return this.orm.call(
+            this.props.record.resModel,
+            'action_configure_with_optional_install',
+            [data],
+        ).then(() => self._activateFirstSync(self, data.calendar_select));
+    }
+
+    /**
+     * Activates the external sync for the first time, when the relevant submodule
+     * was just installed and the app not yet refreshed.
+     *
+     * @param {ConfigureExternalSyncWizardFormRenderer} form Renderer
+     * @param {String} calendar one of google/microsoft
+     * @private
+     */
+    async _activateFirstSync(form, calendar) {
+        const calendar_params = CALENDAR_CONFIGS[calendar];
+        // See google/microsoft_calendar for the origin of this shortened version
+        return form.orm.call(
+            'res.users',
+            calendar_params.restart_sync_method,
+            [[session.uid]]
+        ).then(() => form.rpc(
+            calendar_params.sync_route, {
+                model: 'calendar.event',
+                fromurl: window.location.href,
+            }
+        )).then((o) => {
+            if (o.status === "need_auth") {
+                form._before_finally()
```

Comment:
Would give a better name than "before_finally".

When is this hook exactly? Before finishing the sync configuration?
Name should be based on that.

(Also, camelCase ^^)